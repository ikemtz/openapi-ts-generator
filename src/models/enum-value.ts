export interface IEnumValue {
  key?: number;
  name: string;
  titleName: string;
  snakeCaseName: string;
}
