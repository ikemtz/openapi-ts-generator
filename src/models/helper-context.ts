import { IEntity } from './template-data';

export interface IHelperContext {
  data: {
    root: IEntity;
    key: number;
    index: number;
    first: boolean;
  };
  name: string;
}

export type PropertyType = 'valueProperties' | 'referenceProperties';
