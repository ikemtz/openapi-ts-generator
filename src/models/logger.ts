export interface ILogger {
  log: (data: string) => void;
  error: (data: string) => void;
  warn: (data: string) => void;
}

export class MockConsoleLogger implements ILogger {
  // tslint:disable-next-line: no-empty
  public log = (data: string) => {};
  // tslint:disable-next-line: no-empty
  public error = (data: string) => {};
  // tslint:disable-next-line: no-empty
  public warn = (data: string) => {};
}
