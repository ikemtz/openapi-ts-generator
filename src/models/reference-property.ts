export interface IReferenceProperty {
  name: string;
  snakeCaseName: string;
  referenceTypeName: string;
  typeScriptType: string;
  hasValidators: boolean;
  isSameAsParentTypescriptType: boolean;
  hasMultipleValidators: boolean;
  maxItems?: number;
  minItems?: number;
  maxLength?: number;
  minLength?: number;
  maximum?: number;
  minimum?: number;
  isArray: boolean;
  required: boolean;
  isEnum?: boolean;
  initialValue: string;
  initialTestValue: string;
}
