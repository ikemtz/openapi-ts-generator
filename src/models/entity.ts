import { IReferenceProperty } from './reference-property';
import { IValueProperty } from './value-property';
import { IImportType } from './template-data';

export interface IEntity {
  isEnum?: boolean;
  enumValues: (string | { key?: number; name: string; titleName: string })[];
  name: string;
  kebabCasedName: string;
  camelSingularName: string;
  description?: string;
  importTypes: IImportType[];
  valueProperties: IValueProperty[];
  referenceProperties: IReferenceProperty[];
}
