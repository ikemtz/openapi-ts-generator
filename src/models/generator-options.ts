/* eslint-disable @typescript-eslint/no-unused-vars */
import { resolve } from 'path';
import { ILogger } from './logger';
import { IEntity } from './entity';
import { IReferenceProperty } from './reference-property';
import { IValueProperty } from './value-property';

export interface IGeneratorOptions {
  logger?: ILogger;
  outputPath: string;
  openApiJsonUrl?: string;
  openApiJsonFileName?: string;
  genAngularFormGroups?: boolean;
  typeFilterCallBack?: (entity: IEntity, index: number, array: IEntity[]) => boolean;
  valuePropertyTypeFilterCallBack?: (valueProperty: IValueProperty, index: number, array: IValueProperty[]) => boolean;
  referencePropertyTypeFilterCallBack?: (referenceProperty: IReferenceProperty, index: number, array: IReferenceProperty[]) => boolean;
  pathUrlFormattingCallBack?: (url: string) => string;
  templates?: ITemplates | null;
}
export interface ITemplates {
  model?: string;
  testObjectFactory?: string;
  form?: string;
  formGroupFactory?: string;
  modelProperties?: string;
  barrel?: string;
  enum?: string;
  endpoints?: string;
}
export function defaultFilter(
  _value: IEntity | IValueProperty | IReferenceProperty,
  _index: number,
  _array: IEntity[] | IValueProperty[] | IReferenceProperty[],
): boolean {
  return true;
}

export function setGeneratorOptionDefaults(options: IGeneratorOptions): IGeneratorOptions {
  const templateFolder = resolve(`${__dirname}/..`, 'templates');
  options.typeFilterCallBack = options.typeFilterCallBack ?? defaultFilter;
  options.valuePropertyTypeFilterCallBack = options.valuePropertyTypeFilterCallBack ?? defaultFilter;
  options.referencePropertyTypeFilterCallBack = options.referencePropertyTypeFilterCallBack ?? defaultFilter;
  options.templates = {
    ...options.templates,
    model: options.templates?.model ?? `${templateFolder}/model.ts.hbs`,
    form: options.templates?.form ?? `${templateFolder}/form.ts.hbs`,
    formGroupFactory: options.templates?.formGroupFactory ?? `${templateFolder}/form-group-factory.ts.hbs`,
    testObjectFactory: options.templates?.testObjectFactory ?? `${templateFolder}/test-object-factory.ts.hbs`,
    barrel: options.templates?.barrel ?? `${templateFolder}/index.ts.hbs`,
    enum: options.templates?.barrel ?? `${templateFolder}/enum.ts.hbs`,
    modelProperties: options.templates?.barrel ?? `${templateFolder}/model-properties.ts.hbs`,
    endpoints: options.templates?.endpoints ?? `${templateFolder}/endpoints.ts.hbs`,
  };
  return options;
}
