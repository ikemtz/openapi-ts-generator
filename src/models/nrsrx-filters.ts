/* eslint-disable @typescript-eslint/no-unused-vars */
import { IEntity } from './entity';
import { IValueProperty } from './value-property';
const edmRegex = /^Edm[A-z]*Kind$/;

export function nrsrxTypeFilterCallBack(val: IEntity, _i: number, _arr: IEntity[]): boolean {
  return !val.name.endsWith('ODataEnvelope') && !edmRegex.test(val.name);
}
export function nrsrxValuePropertyTypeFilterCallBack(val: IValueProperty, _i: number, _arr: IValueProperty[]): boolean {
  return !val.name.startsWith('created') && !val.name.startsWith('updated');
}
