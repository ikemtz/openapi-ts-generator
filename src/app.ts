import { generateTsModels } from '.';
import { IGeneratorOptions } from './models/generator-options';
import { nrsrxTypeFilterCallBack, nrsrxValuePropertyTypeFilterCallBack } from './models/nrsrx-filters';
import { IEntity } from './models/template-data';

export const unitGenerationOptions = {
  openApiJsonUrl: 'https://d-unit-wal-00-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './output_unit/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};
export const messageGenerationOptions = {
  openApiJsonUrl: 'https://d-msng-wal-01-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './output_msng/',
  typeFilterCallBack: (val: IEntity, i: number, arr: IEntity[]) =>
    nrsrxTypeFilterCallBack(val, i, arr) && val.name !== 'GetMessageInfoResponse',
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};
export const accountGenerationOptions = {
  openApiJsonUrl: 'https://d-acct-wal-01-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './output_acct/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};

export const fileEmployeeGenerationOptions: IGeneratorOptions = {
  openApiJsonFileName: '../src/test-swagger-document.json',
  outputPath: './output_empl/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};

export function generateUnitFiles(options: IGeneratorOptions = unitGenerationOptions) {
  return generateTsModels(options);
}

export function generateMessageFiles(options: IGeneratorOptions = messageGenerationOptions) {
  return generateTsModels(messageGenerationOptions);
}

export function generateAccountFiles(options: IGeneratorOptions = accountGenerationOptions) {
  return generateTsModels(accountGenerationOptions);
}

export function generateEmployeeFiles(options: IGeneratorOptions = fileEmployeeGenerationOptions) {
  return generateTsModels(fileEmployeeGenerationOptions);
}

generateUnitFiles();
generateMessageFiles();
generateAccountFiles();
generateEmployeeFiles();
