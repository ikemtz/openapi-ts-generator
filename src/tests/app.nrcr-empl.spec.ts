import { mkdirSync } from 'fs';
import { generateTsModels, nrsrxTypeFilterCallBack, nrsrxValuePropertyTypeFilterCallBack } from '..';
import { ValidateFiles } from './app.spec';
import { IGeneratorOptions } from '../models/generator-options';

const fileEmployeeGenerationOptionsFactory = (): IGeneratorOptions => ({
  openApiJsonFileName: '../src/open-api-spec-docs/nrcrn-empl.json',
  outputPath: './jest_output/empl/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
  genAngularFormGroups: true,
});

describe('File Based - Full Integration Tests', () => {
  describe('NRSRx Employee OData Microservice', () => {
    const options = fileEmployeeGenerationOptionsFactory();
    it('should generate files', async () => {
      try {
        mkdirSync(options.outputPath, { recursive: true });
      } catch {
        // ignore
      }
      await generateTsModels(options);
      ValidateFiles(options);
    });
  });
});
