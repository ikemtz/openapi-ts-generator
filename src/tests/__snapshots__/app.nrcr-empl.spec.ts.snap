// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files 1`] = `
Array [
  "employee-certification.form-group-fac.ts",
  "employee-certification.model.ts",
  "employee-certification.properties.ts",
  "employee-competency.form-group-fac.ts",
  "employee-competency.model.ts",
  "employee-competency.properties.ts",
  "employee-health-item.form-group-fac.ts",
  "employee-health-item.model.ts",
  "employee-health-item.properties.ts",
  "employee.form-group-fac.ts",
  "employee.model.ts",
  "employee.properties.ts",
  "endpoints.ts",
  "index.ts",
]
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee.form-group-fac.ts 1`] = `
"/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export interface IEmployeeForm {
    id: new FormControl<string | null>;
    lastName: new FormControl<string>;
    firstName: new FormControl<string>;
    birthDate: new FormControl<Date | null>;
    mobilePhone: new FormControl<string | null>;
    homePhone: new FormControl<string | null>;
    photo: new FormControl<string | null>;
    email: new FormControl<string>;
    addressLine1: new FormControl<string | null>;
    addressLine2: new FormControl<string | null>;
    city: new FormControl<string | null>;
    state: new FormControl<string | null>;
    zip: new FormControl<string | null>;
    isEnabled: new FormControl<boolean>;
    hireDate: new FormControl<Date>;
    fireDate: new FormControl<Date | null>;
    totalHoursOfService: new FormControl<number | null>;
    certificationCount: new FormControl<number>;
    competencyCount: new FormControl<number>;
    healthItemCount: new FormControl<number>;
    employeeCertifications: new FormArray;
    employeeCompetencies: new FormArray;
    employeeHealthItems: new FormArray;
}

export function EmployeeFormGroupFac(): FormGroup<IEmployeeForm> {
  return new FormGroup<IEmployeeForm>({
    id: new FormControl<string | null>(null),
    lastName: new FormControl<string>('', { validators: Validators.compose([Validators.required, Validators.maxLength(250)]), nonNullable: true }),
    firstName: new FormControl<string>('', { validators: Validators.compose([Validators.required, Validators.maxLength(250)]), nonNullable: true }),
    birthDate: new FormControl<Date | null>(null),
    mobilePhone: new FormControl<string | null>(null, { validators: Validators.maxLength(25) } ),
    homePhone: new FormControl<string | null>(null, { validators: Validators.maxLength(25) } ),
    photo: new FormControl<string | null>(null, { validators: Validators.maxLength(4000) } ),
    email: new FormControl<string>('', { validators: Validators.compose([Validators.required, Validators.maxLength(250)]), nonNullable: true }),
    addressLine1: new FormControl<string | null>(null, { validators: Validators.maxLength(250) } ),
    addressLine2: new FormControl<string | null>(null, { validators: Validators.maxLength(250) } ),
    city: new FormControl<string | null>(null, { validators: Validators.maxLength(150) } ),
    state: new FormControl<string | null>(null, { validators: Validators.compose([Validators.minLength(2), Validators.maxLength(2), Validators.pattern('^[A-Z]{2}$')]) }),
    zip: new FormControl<string | null>(null, { validators: Validators.maxLength(10) } ),
    isEnabled: new FormControl<boolean>(false, { validators: Validators.required, nonNullable: true } ),
    hireDate: new FormControl<Date>(new Date(), { validators: Validators.required, nonNullable: true } ),
    fireDate: new FormControl<Date | null>(null),
    totalHoursOfService: new FormControl<number | null>(null, { validators: Validators.min(0) } ),
    certificationCount: new FormControl<number>(0, { validators: Validators.compose([Validators.required, Validators.max(99)]), nonNullable: true }),
    competencyCount: new FormControl<number>(0, { validators: Validators.compose([Validators.required, Validators.min(0), Validators.max(99)]), nonNullable: true }),
    healthItemCount: new FormControl<number>(0, { validators: Validators.compose([Validators.required, Validators.min(0), Validators.max(99)]), nonNullable: true }),
    employeeCertifications: new FormArray([]),
    employeeCompetencies: new FormArray([]),
    employeeHealthItems: new FormArray([]),
  });
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee.model.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */

import { IEmployeeCertification } from './employee-certification.model';
import { IEmployeeCompetency } from './employee-competency.model';
import { IEmployeeHealthItem } from './employee-health-item.model';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IEmployee {
  id?: string;
  lastName?: string;
  firstName?: string;
  birthDate?: Date;
  mobilePhone?: string;
  homePhone?: string;
  photo?: string;
  email?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  isEnabled?: boolean;
  hireDate?: Date;
  fireDate?: Date;
  totalHoursOfService?: number;
  certificationCount?: number;
  competencyCount?: number;
  healthItemCount?: number;
  employeeCertifications?: IEmployeeCertification[];
  employeeCompetencies?: IEmployeeCompetency[];
  employeeHealthItems?: IEmployeeHealthItem[];
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee.properties.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum EmployeeProperties {
  ID = 'id',
  LAST_NAME = 'lastName',
  FIRST_NAME = 'firstName',
  BIRTH_DATE = 'birthDate',
  MOBILE_PHONE = 'mobilePhone',
  HOME_PHONE = 'homePhone',
  PHOTO = 'photo',
  EMAIL = 'email',
  ADDRESS_LINE_1 = 'addressLine1',
  ADDRESS_LINE_2 = 'addressLine2',
  CITY = 'city',
  STATE = 'state',
  ZIP = 'zip',
  IS_ENABLED = 'isEnabled',
  HIRE_DATE = 'hireDate',
  FIRE_DATE = 'fireDate',
  TOTAL_HOURS_OF_SERVICE = 'totalHoursOfService',
  CERTIFICATION_COUNT = 'certificationCount',
  COMPETENCY_COUNT = 'competencyCount',
  HEALTH_ITEM_COUNT = 'healthItemCount',
  EMPLOYEE_CERTIFICATIONS = 'employeeCertifications',
  EMPLOYEE_COMPETENCIES = 'employeeCompetencies',
  EMPLOYEE_HEALTH_ITEMS = 'employeeHealthItems',
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee-certification.form-group-fac.ts 1`] = `
"/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export interface IEmployeeCertificationForm {
    id: new FormControl<string | null>;
    employeeId: new FormControl<string | null>;
    certificationId: new FormControl<string | null>;
    certificationName: new FormControl<string | null>;
    expiresOnUtc: new FormControl<Date | null>;
}

export function EmployeeCertificationFormGroupFac(): FormGroup<IEmployeeCertificationForm> {
  return new FormGroup<IEmployeeCertificationForm>({
    id: new FormControl<string | null>(null),
    employeeId: new FormControl<string | null>(null),
    certificationId: new FormControl<string | null>(null),
    certificationName: new FormControl<string | null>(null),
    expiresOnUtc: new FormControl<Date | null>(null),
  });
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee-certification.model.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IEmployeeCertification {
  id?: string;
  employeeId?: string;
  certificationId?: string;
  certificationName?: string;
  expiresOnUtc?: Date;
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee-certification.properties.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum EmployeeCertificationProperties {
  ID = 'id',
  EMPLOYEE_ID = 'employeeId',
  CERTIFICATION_ID = 'certificationId',
  CERTIFICATION_NAME = 'certificationName',
  EXPIRES_ON_UTC = 'expiresOnUtc',
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee-competency.form-group-fac.ts 1`] = `
"/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export interface IEmployeeCompetencyForm {
    id: new FormControl<string | null>;
    employeeId: new FormControl<string | null>;
    competencyId: new FormControl<string | null>;
    competencyName: new FormControl<string | null>;
    expiresOnUtc: new FormControl<Date | null>;
    isEnabled: new FormControl<boolean | null>;
}

export function EmployeeCompetencyFormGroupFac(): FormGroup<IEmployeeCompetencyForm> {
  return new FormGroup<IEmployeeCompetencyForm>({
    id: new FormControl<string | null>(null),
    employeeId: new FormControl<string | null>(null),
    competencyId: new FormControl<string | null>(null),
    competencyName: new FormControl<string | null>(null),
    expiresOnUtc: new FormControl<Date | null>(null),
    isEnabled: new FormControl<boolean | null>(null),
  });
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee-competency.model.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IEmployeeCompetency {
  id?: string;
  employeeId?: string;
  competencyId?: string;
  competencyName?: string;
  expiresOnUtc?: Date;
  isEnabled?: boolean;
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee-competency.properties.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum EmployeeCompetencyProperties {
  ID = 'id',
  EMPLOYEE_ID = 'employeeId',
  COMPETENCY_ID = 'competencyId',
  COMPETENCY_NAME = 'competencyName',
  EXPIRES_ON_UTC = 'expiresOnUtc',
  IS_ENABLED = 'isEnabled',
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee-health-item.form-group-fac.ts 1`] = `
"/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export interface IEmployeeHealthItemForm {
    id: new FormControl<string | null>;
    employeeId: new FormControl<string | null>;
    healthItemId: new FormControl<string | null>;
    healthItemName: new FormControl<string | null>;
    expiresOnUtc: new FormControl<Date | null>;
    isEnabled: new FormControl<boolean | null>;
}

export function EmployeeHealthItemFormGroupFac(): FormGroup<IEmployeeHealthItemForm> {
  return new FormGroup<IEmployeeHealthItemForm>({
    id: new FormControl<string | null>(null),
    employeeId: new FormControl<string | null>(null),
    healthItemId: new FormControl<string | null>(null),
    healthItemName: new FormControl<string | null>(null),
    expiresOnUtc: new FormControl<Date | null>(null),
    isEnabled: new FormControl<boolean | null>(null),
  });
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee-health-item.model.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IEmployeeHealthItem {
  id?: string;
  employeeId?: string;
  healthItemId?: string;
  healthItemName?: string;
  expiresOnUtc?: Date;
  isEnabled?: boolean;
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: employee-health-item.properties.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum EmployeeHealthItemProperties {
  ID = 'id',
  EMPLOYEE_ID = 'employeeId',
  HEALTH_ITEM_ID = 'healthItemId',
  HEALTH_ITEM_NAME = 'healthItemName',
  EXPIRES_ON_UTC = 'expiresOnUtc',
  IS_ENABLED = 'isEnabled',
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: endpoints.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #endpoints-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum Endpoints {
  employees = '/odata/v1/Employees',
}
export interface IEndpoints {
  readonly employees: string;
}
"
`;

exports[`File Based - Full Integration Tests NRSRx Employee OData Microservice should generate files: index.ts 1`] = `
"/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './employee-certification.form-group-fac';
export * from './employee-certification.model';
export * from './employee-certification.properties';
export * from './employee-competency.form-group-fac';
export * from './employee-competency.model';
export * from './employee-competency.properties';
export * from './employee-health-item.form-group-fac';
export * from './employee-health-item.model';
export * from './employee-health-item.properties';
export * from './employee.form-group-fac';
export * from './employee.model';
export * from './employee.properties';
"
`;
